### PR TITLE
update polkadot v0.9.33

### PR DESCRIPTION
**Pull request by blasrodri**

### CODE REVIEW

**Comment by blasrodri** [comment]
NOTE: I still need to figure out the changes in the laves of the `Proof` of `Beefy`.

File: contracts/pallet-ibc/rpc/src/lib.rs [hunk 1]
```diff
@@ -598,6 +598,11 @@ where
 	fn query_proof(&self, height: u32, mut keys: Vec<Vec<u8>>) -> Result<Proof> {
 		let api = self.client.runtime_api();
 		let at = BlockId::Number(height.into());
+		let hash_at = self
+			.client
+			.block_hash_from_id(&at)
```
Comment (blasrodri) [inline contracts/pallet-ibc/rpc/src/lib.rs:603]: nice

**Comment by Wizdave97** [comment]
They've made some breaking changes in the updated subxt version, such as redefining  of sp_runtime::traits::Header  in subxt, this complicates some things in the grandpa prover setup.

**Comment by blasrodri** [comment]
> They've made some breaking changes in the updated subxt version, such as redefining of sp_runtime::traits::Header in subxt, this complicates some things in the grandpa prover setup.

ooff

**Comment by Wizdave97** [comment]
I'll make grandpa primitives like beefy primitives, we won't depend on the sp_runtime traits, we'll just use bytes, and decode the concrete values in ics10-grandpa light client.

**Comment by Wizdave97** [comment]
I'll finish up this PR.

**Comment by Wizdave97** [comment]
This is almost done, should be done later today or early tomorrow, I'm fixing some bugs in some implementations as the tests are no longer passing.

**Review by vmarkushin — APPROVED** [review]

File: algorithms/beefy/verifier/src/lib.rs [hunk 2]
```diff
@@ -163,7 +162,15 @@ where
...
-	let leaf_pos = mmr_lib::leaf_index_to_pos(mmr_update.mmr_proof.leaf_index);
+	// We are trying to verify the proof for the latest mmr leaf so we expect the proof to contain a
+	// singular leaf index
+	let leaf_index = if mmr_update.mmr_proof.leaf_indices.len() == 1 {
+		mmr_update.mmr_proof.leaf_indices[0]
+	} else {
+		Err(BeefyClientError::ExpectedSingleLeafIndex)?
+	};
+
+	let leaf_pos = mmr_lib::leaf_index_to_pos(leaf_index);
```
Comment (obsessed-cake) [inline algorithms/beefy/verifier/src/lib.rs:173]:
```suggestion
	let leaf_index = mmr_update
		.mmr_proof
		.leaf_indices
		.get(0)
		.ok_or(BeefyClientError::ExpectedSingleLeafIndex)?;

	let leaf_pos = mmr_lib::leaf_index_to_pos(*leaf_index);
```

File: algorithms/beefy/verifier/src/tests.rs [hunk 3]
```diff
@@ -47,9 +46,12 @@ async fn test_verify_mmr_with_proof() {
 	let para_client = subxt::client::OnlineClient::<PolkadotConfig>::from_url(para_ws_url)
 		.await
 		.unwrap();
-
+	println!("Waiting for parachain to start producing blocks");
+	let block_sub = para_client.blocks().subscribe_finalized().await.unwrap();
```
Comment (obsessed-cake) [inline algorithms/beefy/verifier/src/tests.rs:50]:
```suggestion
	let block_sub = para_client.blocks().subscribe_finalized().await.expect("Can not create a new subscription");
```

**Review by obsessed-cake — APPROVED** [review]

**Review by obsessed-cake — APPROVED** [review]

**Review by Wizdave97 — APPROVED** [review]

**Comment by blasrodri** [comment]
@Wizdave97 do you want @vmarkushin to check this? Or is it good to go?

**Comment by Wizdave97** [comment]
> @Wizdave97 do you want @vmarkushin to check this? Or is it good to go?

it's good to go

**Pull request merged**